catalog/spatial/csw/spatial-csw-source/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/source/CswSource.java

```diff
@@ -388,153 +459,76 @@ public void destroy(int code) {
-        if (coordinateOrder != null) {
-            setCoordinateOrder(coordinateOrder);
-            configuration.put(CSW_AXIS_ORDER_PROPERTY, cswSourceConfiguration.getCswAxisOrder());
-        }
-
-        Boolean posListProp = (Boolean) configuration.get(USE_POS_LIST_PROPERTY);
-        if (posListProp != null) {
-            cswSourceConfiguration.setUsePosList(posListProp);
-        }
-
+        // Set Blank Defaults
         String spatialFilter = (String) configuration.get(FORCE_SPATIAL_FILTER_PROPERTY);
         if (StringUtils.isBlank(spatialFilter)) {
             spatialFilter = NO_FORCE_SPATIAL_FILTER;
         }
-        forceSpatialFilter = spatialFilter;
-
-        String createdProp = (String) configuration.get(CREATED_DATE_MAPPING_PROPERTY);
-        if (StringUtils.isNotBlank(createdProp)) {
-            cswSourceConfiguration.setCreatedDateMapping(createdProp);
-        }
+        setForceSpatialFilter(spatialFilter);
 
-        String effectiveProp = (String) configuration.get(EFFECTIVE_DATE_MAPPING_PROPERTY);
-        if (StringUtils.isNotBlank(effectiveProp)) {
-            cswSourceConfiguration.setEffectiveDateMapping(effectiveProp);
+        String currentContentTypeMapping =
+                (String) configuration.get(CONTENT_TYPE_MAPPING_PROPERTY);
+        if (StringUtils.isBlank(currentContentTypeMapping)) {
+            cswSourceConfiguration.setContentTypeMapping(CswRecordMetacardType.CSW_TYPE);
         }
-
-        String modifiedProp = (String) configuration.get(MODIFIED_DATE_MAPPING_PROPERTY);
-        if (StringUtils.isNotBlank(modifiedProp)) {
-            cswSourceConfiguration.setModifiedDateMapping(modifiedProp);
-        }
-
-        String currentContentTypeMapping = ((String) configuration
-                .get(CONTENT_TYPE_MAPPING_PROPERTY));
-
-        if (StringUtils.isNotBlank(currentContentTypeMapping)) {
-            String previousContentTypeMapping = cswSourceConfiguration.getContentTypeMapping();
-            LOGGER.debug("{}: Previous content type mapping: {}.", cswSourceConfiguration.getId(),
-                    previousContentTypeMapping);
-            currentContentTypeMapping = currentContentTypeMapping.trim();
-            if (!currentContentTypeMapping.equals(previousContentTypeMapping)) {
-                LOGGER.debug("{}: The content type has been updated from {} to {}.",
-                        cswSourceConfiguration.getId(), previousContentTypeMapping,
-                        currentContentTypeMapping);
-                contentTypes.clear();
+        // Filter Configuration Map
+        Map<String, Object> configs = filter(configuration);
+
+        // Run Filtered Consumers
+        for (Map.Entry<String, Object> entry : configs.entrySet()) {
+            String s = entry.getKey();
+            Consumer c = consumerMap.get(s);
+            if (c != null) {
+                c.accept(configs.get(s));
```

Comment:
You can use `entry.getValue()` instead of `configs.get(s)`. Also, it would probably be better to use variable names other than `c` and `s`.